core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -1370,4 +1370,12 @@ default boolean isSupportedVersionType(ConnectorSession session, SchemaTableName
     {
         throw new TrinoException(NOT_SUPPORTED, "This connector does not support versioned tables");
     }
+
+    /**
+     * Checks if a table exists.
+     */
+    default boolean tableExists(ConnectorSession session, SchemaTableName tableName)
```

Comment:
After having https://github.com/trinodb/trino/pull/11555 merged, please do add, as part of this commit, the usage of this method in `DropViewTask`, `DropMaterializedViewTask`.

If you lookup in the code for the usages of the method `io.trino.spi.connector.ConnectorMetadata#getTableHandle(io.trino.spi.connector.ConnectorSession, io.trino.spi.connector.SchemaTableName)` you will notice a few places containing checks like:

```
metadata.getTableHandle(session, name) == null
```

which can obviously be changed to use this newly introduced method.